File: kotlin/DesignPattern/src/main/kotlin/abstractfactorypattern/source/EUBeverageFactory.kt
package abstractfactorypattern.source

class EUBeverageFactory: BeverageFactory {

    override fun createBlackTea(): BlackTea {
        return EarlGreyBlackTea()
    }

    override fun createGreenTea(): GreenTea {
        return SenchaGreenTea()
    }
}

File: swift/DesignPattern.playground/Pages/AbstractFactoryPattern.xcplaygroundpage/Sources/EUBeverageFactory.swift
import Foundation

open class EUBeverageFactory: BeverageFactory {
    
    public init() {}
    
    public func createBlackTea() -> BlackTea? {
        return EarlGreyBlackTea()
    }
    
    public func createGreenTea() -> GreenTea? {
        return SenchaGreenTea()
    }
}


